scripts/SetGateVariable.ps1

```diff
@@ -14,25 +14,20 @@ Write-Host "##vso[task.setvariable variable=IsEnabled;isOutput=true]False"
 $path = "$($env:System_DefaultWorkingDirectory)/_Build Image Release Artefacts/aws-$BaseImageName/$BaseImageName.txt"
 Write-Host "Using $path"
 if (Test-Path $path) {
-    try{
-        $amiID = Get-Content -Path $path
-        Write-Host "AMI ID $($amiID)"
-        $imageName = (Get-EC2Image -ImageId $amiID).Name
-        $imageName -match "$BaseImageName[-]?[0-9]+"
-        $version = $Matches[0]
-        Write-Host "Version : $version"
-        $stackname = "$version-$stackname"
-        Write-Host "Stack name : $stackname"
-        #Set Variables
-        Write-Host "##vso[task.setvariable variable=stack;isOutput=true]$stackname"
-        Write-Host "##vso[task.setvariable variable=version;isOutput=true]$version"
-        Write-Host "##vso[task.setvariable variable=ImageID;isOutput=true]$amiID"
-        Write-Host "##vso[task.setvariable variable=IsEnabled;isOutput=true]True"
-        Write-host "The value of Variable IsEnabled is updated to True and output variable ImageID to $amiID"
-    } catch{
-        $_ | Out-Default | Write-Host
-        Throw "Failed to set Task Variable"
-    }
```

Comment:
Please revert this code. I've said it before, if its useful for debugging, its useful to leave in. Much more obvious to have an exception thrown  than a silent failure.

Reply:
I have reverted the code and debugging the failure of SetGateVariable task which says invalid AMI ID,though it is available in AWS
https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_releaseProgress?_a=release-environment-logs&releaseId=251&environmentId=3509&jobTimelineRecordIdToSelect=3028c7e8-df94-5918-0c54-a70bb390dc4d&selectTaskWithIndex=6#